pkg/vsphere/disk/disk_manager.go

```diff
@@ -282,3 +296,37 @@ func (m *Manager) devicePathByURI(ctx context.Context, datastoreURI string) (str
 
 	return fmt.Sprintf(m.byPathFormat, *disk.UnitNumber), nil
 }
+
+//Utility Functions
+
+// configureDeviceSpec is a function that provides the spec for our new volume. We do this because we need to be able to retry on the actual reconfigure task. Whic is hidden in govmomi calls unless we call the reconfigure operation  ourselves.
+func configureDeviceSpec(ctx context.Context, v object.VirtualMachine, op types.VirtualDeviceConfigSpecOperation, fop types.VirtualDeviceConfigSpecFileOperation, devices ...types.BaseVirtualDevice) types.VirtualMachineConfigSpec {
```

Comment:
Looks like you should be able to use VirtualDeviceList.ConfigSpec() instead.  Not sure why VirtualMachine.configureDevice() isn't using VirtualDeviceList.ConfigSpec() too, might just be a missed cleanup - I'll look into it.